sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/util/HashString.java

```diff
@@ -0,0 +1,25 @@
+package org.dbsp.util;
+
+import javax.annotation.Nullable;
+
+/** A Hash value produced by SHA256 */
+public record HashString(String value) {
+    /** Number of digits that is safe to use in abbreviations */
+    static final int SHORT_SIZE = 12;
```

Comment:
12 seems only just sufficient to prevent hash collisions, would maybe 16 be a better margin?